lib/shanty_docker_plugin/docker_credentials.rb

```diff
@@ -0,0 +1,47 @@
+require 'base64'
+require 'docker'
+require 'json'
+require 'uri'
+
+module DockerPlugin
+  # Public: Set up credentials from ~/.dockercfg
+  class DockerCredentials
+    def initialize(dockercfg_file = "#{Dir.home}/.dockercfg")
+      @dockercfg_file = dockercfg_file
+    end
+
+    def auth!(registry)
+      Docker.creds = find_credentials(registry)
+    end
+
+    private
+
+    def find_credentials(registry)
+      credentials[registry] || credentials['default'] || {}
+    end
+
+    def credentials
+      @credentials ||= read_dockercfg.each_with_object({}) do |v, acc|
```

Comment:
Change this slightly to capture `k` and `v` instead of just `v`:

```ruby
@credentials ||= read_dockercfg.each_with_object({}) do |(k, v), acc|
```